src/module.c

```diff
@@ -1535,8 +1595,65 @@ int RM_ReplyWithEmptyArray(RedisModuleCtx *ctx) {
     return REDISMODULE_OK;
 }
 
+/* Reply to the client with an empty map.
+ *
+ * The function always returns REDISMODULE_OK. */
+int RM_ReplyWithEmptyMap(RedisModuleCtx *ctx) {
+    client *c = moduleGetReplyClient(ctx);
+    if (c == NULL) return REDISMODULE_OK;
+    addReply(c,shared.emptymap[c->resp]);
+    return REDISMODULE_OK;
+}
+
+/* Reply to the client with an empty set.
+ *
+ * The function always returns REDISMODULE_OK. */
+int RM_ReplyWithEmptySet(RedisModuleCtx *ctx) {
+    client *c = moduleGetReplyClient(ctx);
+    if (c == NULL) return REDISMODULE_OK;
+    addReply(c,shared.emptyset[c->resp]);
+    return REDISMODULE_OK;
+}
+
+void moduleReplySetCollectionLength(RedisModuleCtx *ctx, long len, int type) {
+    client *c = moduleGetReplyClient(ctx);
+    if (c == NULL) return;
```

Comment:
Can this even happened?

Reply:
@MeirShpilraien technically the function can return null.